R/websocket.R

```diff
@@ -6,6 +6,56 @@ NULL
 # Used to "null out" handler functions after a websocket client is closed
 null_func <- function(...) { }
 
+options <- list(max_message_size=32*1024*1024)
+#' Manage WebSocket options
+#'
+#' Lists the default options or assists in creating a custom configuration to
+#' use when creating a new WebSocket.
+#'
+#' @param ... Named options to be overridden.
+#' @details
+#' The following options are available:
+#' \describe{
+#'   \item{\code{max_message_size}}{The maximum size of a message in bytes. If a message
+#'   larger than this is sent, the connection will fail with the \code{message_too_big}
+#'   protocol error.}
+#' }
+#' @return If \code{opts} is \code{NULL} (the default), returns the default options. Otherwise,
+#' returns a named list that can be passed into \code{WebSocket$new()} as the \code{wsOptions}
+#' parameter to customize the behavior of a WebSocket.
+#' @examples
+#' ws <- WebSocket$new("ws://echo.websocket.org/", opts = wsOptions(max_message_size=1234))
+#' @export
+wsOptions <- function(...){
+  opts <- list(...)
```

Comment:
What's the reasoning for not having an explicit arg for each option here?

Reply:
Fair point. The thought was that packages like `curl` have hundreds of options and so enumerating them all would get overwhelming. 

I don't see us getting up to that many, but there are a dozen+ options that we might eventually expose.